helpers/kernel_config.go

```diff
@@ -174,53 +175,71 @@ var KernelConfigKeyIDToString = map[KernelConfigOption]string{
 type KernelConfig struct {
 	configs         map[KernelConfigOption]interface{} // predominantly KernelConfigOptionValue, sometimes string
 	needed          map[KernelConfigOption]interface{}
-	KConfigFilePath string
+	kConfigFilePath string
 }
 
 // InitKernelConfig inits external KernelConfig object
 func InitKernelConfig() (*KernelConfig, error) {
 	config := KernelConfig{}
 
 	// special case: user provided kconfig file (it MUST exist)
+
 	osKConfigFilePath, err := checkEnvPath("LIBBPFGO_KCONFIG_FILE") // override /proc/config.gz or /boot/config-$(uname -r) if needed (containers)
 	if err != nil {
-		return nil, err
+		return &config, err
 	}
 	if len(osKConfigFilePath) > 2 {
 		if _, err := os.Stat(osKConfigFilePath); err != nil {
-			return nil, err
+			return &config, err
 		}
-		config.KConfigFilePath = osKConfigFilePath // user might need to know used KConfig file in order to override kconfig for libbpf (example)
+		config.kConfigFilePath = osKConfigFilePath
 		if err := config.initKernelConfig(osKConfigFilePath); err != nil {
-			return nil, err
+			return &config, err
 		}
 
 		return &config, nil
 	}
 
 	// fastpath: check config.gz in procfs first
+
 	configGZ := "/proc/config.gz"
 	if _, err1 := os.Stat(configGZ); err1 == nil {
-		config.KConfigFilePath = configGZ // same thing here
+		config.kConfigFilePath = configGZ
 		if err2 := config.initKernelConfig(configGZ); err2 != nil {
-			return nil, err2
+			return &config, err2
 		}
 
 		return &config, nil
 	} // ignore if /proc/config.gz does not exist
 
 	// slowerpath: /boot/$(uname -r)
+
 	releaseVersion, err := UnameRelease()
 	if err != nil {
-		return nil, err
+		return &config, err
 	}
+
 	releaseFilePath := fmt.Sprintf("/boot/config-%s", releaseVersion)
-	config.KConfigFilePath = releaseFilePath // and here
-	if err := config.initKernelConfig(releaseFilePath); err != nil {
-		return nil, err
+	config.kConfigFilePath = releaseFilePath
+	err = config.initKernelConfig(releaseFilePath)
+
+	return &config, err
+}
+
+// GetKernelConfigFilePath gives the kconfig file chosen by InitKernelConfig during initialization
+func (k *KernelConfig) GetKernelConfigFilePath() string {
+	return k.kConfigFilePath
+}
+
+// AddCustomKernelConfigs allows user to extend list of possible existing kconfigs to be parsed from kConfigFilePath
+func (k *KernelConfig) AddCustomKernelConfigs(values map[KernelConfigOption]string) error {
```

Comment:
I'd prefer this method to just take a key and value, let the user iterate over their map and call it. 

Also, what exactly is the usecase for it? And should we have validation for the keys being real config options? Is it to edit existing keys or add new ones? If it's to add new ones we should validate that it doesn't already exist.

Reply:
There is no need for validation as the values can indeed be Undefined because they don't exist in running kconfig file but they could exist in some others. We would have to have an updated list of possible config options for new kernels as well and it does no make much sense. 

User has to inform option and if it exists it's value will be defined, if not.. undefined.

Reply:
About use case and iteration, could you check my pull request to Tracee as a consumer to this and see if you still think that is the best way ? Thanks

Reply:
Iteration with single items would make it equal to initial logic of pushing needed config options one by one.. maybe it's the best way forward indeed .. to keep everything similar.